lib/java/com/google/android/material/appbar/MaterialToolbar.java

```diff
@@ -378,16 +378,19 @@ public boolean isSubtitleCentered() {
 
   private void initBackground(Context context) {
     Drawable background = getBackground();
-    if (background != null && !(background instanceof ColorDrawable)) {
-      return;
+    ColorStateList backgroundCSL = DrawableUtils.getColorStateListOrNull(background);
+
+    if (background == null || backgroundCSL != null) {
+      MaterialShapeDrawable materialShapeDrawable = new MaterialShapeDrawable();
+      if (backgroundCSL != null) {
+        materialShapeDrawable.setFillColor(backgroundCSL);
+      } else {
+        materialShapeDrawable.setFillColor(ColorStateList.valueOf(Color.TRANSPARENT));
+      }
```

Comment:
```suggestion
      if (backgroundCSL == null) {
        backgroundCSL = ColorStateList.valueOf(Color.TRANSPARENT);
      }
      materialShapeDrawable.setFillColor(backgroundCSL);
```

Reply:
If it's about conciseness, we can do it this way:

```java
// Add a MaterialShapeDrawable as background that supports tinting in every API level.
Drawable background = getBackground();
ColorStateList backgroundCSL = background != null
    ? DrawableUtils.getColorStateListOrNull(background)
    : ColorStateList.valueOf(Color.TRANSPARENT);

if (backgroundCSL != null) {
  MaterialShapeDrawable materialShapeDrawable = new MaterialShapeDrawable();
  materialShapeDrawable.setFillColor(backgroundCSL);
  materialShapeDrawable.initializeElevationOverlay(context);
  materialShapeDrawable.setElevation(ViewCompat.getElevation(this));
  ViewCompat.setBackground(this, materialShapeDrawable);
}
```

Reply:
Even better!

Reply:
Ok, done.